python/ray/autoscaler/_private/gcp/node_provider.py

```diff
@@ -274,6 +276,34 @@ def _get_cached_node(self, node_id: str) -> GCPNode:
     def bootstrap_config(cluster_config):
         return bootstrap_gcp(cluster_config)
 
+    @staticmethod
+    def fillout_available_node_types_resources(
+        cluster_config: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Fill out TPU resources to the cluster config."""
+        if "available_node_types" not in cluster_config:
+            return cluster_config
+        cluster_config = copy.deepcopy(cluster_config)
+        available_node_types = cluster_config["available_node_types"]
+        for node_type in available_node_types:
+            node_config = available_node_types[node_type]["node_config"]
+            if get_node_type(node_config) == GCPNodeType.TPU:
+                autodetected_resources = {}
+                accelerator_type = ""
+                if "acceleratorType" in node_config:
+                    accelerator_type = node_config["acceleratorType"]
+                elif "acceleratorConfig" in node_config:
+                    accelerator_type = tpu_accelerator_config_to_type(
+                        node_config["acceleratorConfig"]
+                    )
+                if not accelerator_type:
+                    continue
+                autodetected_resources[f"TPU-{accelerator_type}"] = 1
```

Comment:
I intentionally left out `tpu_id` resource here because I don't believe it can be populated at the `fillout_available_node_types_resources` stage. The VM(s) will have to be created already.

Yes, for autoscaling, the user will make a request of the TPU accelerator type, e.g. v4-16, in which case the cluster launcher will know which node from the config to create. So the TPU pod is a single upscale/downscale unit.